### DISCUSSION TITLE
How can I create a new note while prompting for the file type?

### DISCUSSION THREAD

**Discussion by jgarte** [localauthor/zk#70] [Q&A]
Hi,

What is the recommended way to create a new note while prompting for the file type?

For example, the default file type is `.txt` and I want to create a file of type `.org`.

**Comment by localauthor** [comment]
Hello! ZK isn't set up for working with files with different extensions, since the variable `zk-file-extension` must be set to a single extension, as a string. Maybe there's a way to accomplish what you want using files with the same type, if you can describe your use case?

**Reply by jgarte** [reply]
Hi,

I think denote has a way to do this. 

https://git.sr.ht/~protesilaos/denote/tree/main/item/denote.el

I've forgotten the exact denote function name that I used but it was definitely a `denote-*` function that also renamed the file extension.

I've switched to zk now 🙃 

I'll try to do some reading/trying and get back to you on that.

**Reply by localauthor** [reply]
Cool, let me know what you find. Happy to help if I can.

Side question: what made you switch from denote? I haven’t used it, so I’m just curious about what differences you see between this and that.

**Reply by jgarte** [reply]
The simplicity with which you explained backlinks in your video (documentation) and how the "UI" is set up to interacts with backlinks were factors. The video documentaton is good and short. zk video [shorts](https://en.wikipedia.org/wiki/YouTube_Shorts) might be a nice idea.

- The code base is smaller.

- I also like that you encourage forking 

- I'm using `zk` as a stepping stone to maybe roll my own or just learn some more elisp along the way.

- I might take some function ideas from denote for use in zk (for example, the file type rename mentioned here elsewhere)

It is also very early for me with `zk`. I started using it yesterday.

But I quickly integrated zk into my Guix Home/System setup:

https://git.sr.ht/~whereiseveryone/confetti/tree/s/item/conf/confetti/features/zk.scm

**Reply by localauthor** [reply]
Cool! Yeah, the aim here is not to over-complicate things :)

**Reply by jgarte** [reply]
> not to over-complicate things :)

I'll try... ;()

**Reply by maasdorp** [reply]
I also like the simplicity of zk and for my academic use-case I think the index and desktop functions are the most important differentiators compared to other tools.

**Reply by localauthor** [reply]
That's really helpful to hear, thanks. I happen to think zk-index is the real killer feature (imho ;) )
